File: v131016/TeeNew/Commander_Custom.pas
unit Commander_Custom;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  QButtons,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  Buttons,
  {$ENDIF}
  Base, TeeComma, TeeProcs, TeEngine, Chart;

type
  TCommanderCustom = class(TBaseForm)
    TeeCommander1: TTeeCommander;
    SpeedButton1: TSpeedButton;
    procedure FormCreate(Sender: TObject);
    procedure SpeedButton1Click(Sender: TObject);
  private
    { Private declarations }
    procedure Button1Click(Sender: TObject);
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeCommanderEditor;

procedure TCommanderCustom.FormCreate(Sender: TObject);
var tmpButton : TSpeedButton;
begin
  inherited;
  TeeCommander1.CreateControls([tcbNormal, tcbRotate, tcbSeparator, tcb3D ]);
  { add a new custom button to TeeCommader }
  tmpButton:=TeeCommander1.CreateButton(100, Button1Click, 'Customize','',0);

  { set the custom button bitmap }
  tmpButton.Glyph.Assign(SpeedButton1.Glyph);
  tmpButton.NumGlyphs:=SpeedButton1.NumGlyphs; { 2 images in button }
end;

procedure TCommanderCustom.Button1Click(Sender: TObject);
begin
  TeeCommanderEdit(Self,TeeCommander1);
end;

procedure TCommanderCustom.SpeedButton1Click(Sender: TObject);
begin
  ShowMessage('(This is an example)');
end;

initialization
  RegisterClass(TCommanderCustom);
end.

File: TeeNew/Builder2009/Commander_Custom.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Commander_Custom.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeComma"
#pragma resource "*.dfm"
TCommanderCustom *CommanderCustom;
//---------------------------------------------------------------------------
__fastcall TCommanderCustom::TCommanderCustom(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TCommanderCustom::FormCreate(TObject *Sender)
{
//  const TeeCommanderControls Buttons[]={ tcbRotate, tcbSeparator, tcb3D };
//  TeeCommander1->CreateControls(EXISTINGARRAY(Buttons));
}
//---------------------------------------------------------------------------


void __fastcall TCommanderCustom::ButtonClick(TObject *Sender)
{
  ShowMessage("Hello Custom Button !");
}

void __fastcall TCommanderCustom::FormShow(TObject *Sender)
{
  // add a new custom button to TeeCommander
  TSpeedButton *tmpButton=TeeCommander1->CreateButton(100, &ButtonClick, "Customize","",0);

  // set the custom button bitmap
  tmpButton->Glyph->Assign(SpeedButton1->Glyph);
  tmpButton->NumGlyphs=SpeedButton1->NumGlyphs; // 2 images in button
}
//---------------------------------------------------------------------------

